File: app/src/main/java/com/processdumper/repository/ProcessRepository.java
package com.processdumper.repository;

import android.content.Context;
import android.util.Log;

import com.processdumper.components.MapItem;
import com.processdumper.model.ProcessInfo;
import com.processdumper.model.mapinfo.MapInfo;
import com.processdumper.utils.LogManager;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

public class ProcessRepository {
    private final HashMap<String,ProcessInfo> processes = new HashMap<>();

    public HashMap<String, ProcessInfo> getProcesses() {
        return processes;
    }

    public ProcessInfo getProcessInfo(String packageName){
        return processes.get(packageName);
    }

    public List<MapItem> getAllFormated(){
        List<MapItem> processFormatted = new ArrayList<>();

        for (ProcessInfo process : this.processes.values()) {
            if(process.appName != "App Name not found"){
                processFormatted.add(new MapItem("", process.packageName));
            }
        }

        return processFormatted;
    }

    public boolean add(Context context, String packageName, Integer pid){
        if(pid <= 0){
            return false;
        }
        ProcessInfo info = processes.get(packageName);
        if(info != null){
            info.pids.add(pid);
        }else {
            processes.put(packageName,
                    new ProcessInfo(context, packageName, new ArrayList<>(List.of(pid))));
            ;
        }
        return true;
    }
}

File: app/src/main/java/com/processdumper/repository/ProcessRepository.kt
package com.processdumper.repository

import android.content.Context
import com.processdumper.model.ProcessInfo

class ProcessRepository  {
    private val processes: MutableList<ProcessInfo> = mutableListOf()

    fun add(context: Context, packageName: String, pid: Int) {
        if(pid <= 0){
            return;
        }
        val existing = processes.find { it ->
            it.PackageName === packageName
        }
        if(existing != null){
            existing.Pids.add(pid)
        }
        else {
            processes.add(
                ProcessInfo(
                    context,
                    packageName,
                    mutableListOf(pid)
                )
            )
        }
    }

    fun getAll(): List<ProcessInfo> = processes

    fun getAllFormated(): MutableList<String>{
        var processFormated : MutableList<String> = mutableListOf();
        for (process in this.processes) {
            processFormated.add(process.PackageName);
        }
        return processFormated;
    }
}